Source/WebCore/rendering/MotionPath.cpp

```diff
@@ -194,19 +194,36 @@ std::optional<Path> MotionPath::computePathForRay(const RayPathOperation& rayPat
     return path;
 }
 
+static FloatRoundedRect offsetRectForData(const MotionPathData& data)
+{
+    auto rect = data.containingBlockBoundingRect;
+    auto shiftedPoint = data.offsetFromContainingBlock;
+    shiftedPoint.scale(-1);
+    rect.setLocation(shiftedPoint);
+    return rect;
+}
+
 std::optional<Path> MotionPath::computePathForBox(const BoxPathOperation&, const TransformOperationData& data)
 {
     if (auto motionPathData = data.motionPathData()) {
         Path path;
-        auto rect = motionPathData->containingBlockBoundingRect;
-        auto shiftedPoint = motionPathData->offsetFromContainingBlock;
-        shiftedPoint.scale(-1);
-        rect.setLocation(shiftedPoint);
-        path.addRoundedRect(rect, PathRoundedRect::Strategy::PreferBezier);
+        path.addRoundedRect(offsetRectForData(*motionPathData), PathRoundedRect::Strategy::PreferBezier);
         return path;
     }
     return std::nullopt;
 }
 
 
+std::optional<Path> MotionPath::computePathForShape(const ShapePathOperation& pathOperation, const TransformOperationData& data)
+{
+    if (auto motionPathData = data.motionPathData()) {
+        auto& shape = pathOperation.basicShape();
+        auto rect = motionPathData->containingBlockBoundingRect.rect();
+        if (!is<BasicShapePath>(shape))
+            rect = offsetRectForData(*motionPathData).rect();
```

Comment:
Excluding path seems arbitrary to me. I would expect the behavior to be consistent for all shapes. Can we check what Firefox & Chrome do here?

Reply:
Yeah interestingly offsetting the rect causes the path tests to fail. Gonna try and understand why.